docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,220 @@
+```
+{
+    "network-prefix-valid",
+    "api-vips-defined"
+}
+```
+
+#### Endpoint - Add a skipped validation.
+
+PUT api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
+PUT api/v2/cluster/<id>/skipped-cluster-validations/api-vips-defined
+
+* Will add an entry to the list of validations to be skipped. This will appropriately de-deuplicated and no error will be returned if duplication is attempted.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete a skipped host validation.
+DELETE api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete a skipped cluster validation.
+DELETE api/v2/cluster/<id>/skipped-cluster-validations/no-ip-collisions-in-network
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete all skipped validations for host.
+DELETE api/v2/cluster/<id>/skipped-host-validations
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete all skipped validations for cluster.
+DELETE api/v2/cluster/<id>/skipped-cluster-validations
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+
+### The Kubernetes API
+
+There has been a suggestion that we should create a KubeAPI API so that validations may be skipped for clients using ACM and MCE
+I am a little unsure which namespace we should put this in, so that's one of the open questions.
+
+As I see it, we could implement this with a CRD like this one.
+```
+cat << EOF | oc apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: SkippedValidations
+metadata:
+  name: spoke-1-skipped-validations
+  namespace: spoke-1
+spec:
+  clusterRef:
+   name: spoke-1
+   namespace: spoke-1
+  skipped-host-validations:
+    - no-ip-collisions-in-network
+    - sufficient-network-latency-requirement-for-role
+  skipped-cluster-validations:
+    - network-prefix-valid
+    - api-vips-defined
+EOF
+```
+
+In any case, I think the API should work around this one object.
+
+* If you want to delete all skipped validations, simply delete this CR and await reconciliation.
+* To add or remove a skipped validation, simply edit the CR and wait reconciliation.
+* Perhaps we would stop reconciling this CR once the installation starts as this would make sense.
+
+### The implementation of skipped-validations
+
+There are two ways that are currently proposed.
+
+#### Option 1: Run all validations, allow the user to review but proceed anyway. (Suggested by Michael Hrivnak as an intepretation of requirements)
```

Comment:
This is how I imagined it would work as well.

Reply:
same here.